plugins/hls-refactor-plugin/src/Development/IDE/GHC/ExactPrint.hs

```diff
@@ -430,6 +434,32 @@ graftDecls dst decs0 = Graft $ \dflags a -> do
             | otherwise = DL.singleton (L src e) <> go rest
     modifyDeclsT (pure . DL.toList . go) a
 
+modifySmallestDeclWithM ::
+    forall a.
+    (HasDecls a) =>
+    (SrcSpan -> Bool) ->
+    (LHsDecl GhcPs -> TransformT (Either String) (Maybe [LHsDecl GhcPs])) ->
+    a ->
+    TransformT (Either String) a
+modifySmallestDeclWithM validSpan f a = do
+    let modifyMatchingDecl [] = pure DL.empty
+        modifyMatchingDecl (e@(L src _) : rest)
+            | validSpan $ locA src = do
+                decs' <- fromMaybe [e] <$> f e
```

Comment:
Why does `f` return a `Maybe`? Couldn't it simply return its input untouched?

Reply:
This was all a little silly. There was no need for the Maybe at all, so I reworked the passed-in function.